src/components/Navbar/SearchDrawer/SearchDrawer.tsx

```diff
@@ -88,11 +135,37 @@ const SearchDrawer: React.FC = () => {
         .finally(() => {
           setIsSearching(false);
         });
-    } else {
-      // reset the result
-      setSearchResult(null);
+    },
+    [selectedTranslations],
+  );
+
+  // a ref to know whether this is the initial search request made when the user loads the page or not
+  const isInitialSearch = useRef(true);
```

Comment:
This is related to search page's pagination and should be removed along with everywhere it's being used in.